### PR TITLE
boards/remote-revb: define default MTD device

### PR DESCRIPTION
**Pull request by jdavid**
### Contribution description

Add MTD configuration to the remote-revb board.

### Testing procedure

```
$ BOARD=remote-revb make -C tests/mtd_raw all flash term
[...]
2022-09-20 09:48:12,881 # main(): This is RIOT! (Version: 2022.10-devel-680-gf03f5-remote-revb-mtd)
2022-09-20 09:48:12,881 # Manual MTD test
2022-09-20 09:48:12,882 # init MTD_0… OK (1915 MiB)
> info
2022-09-20 09:48:52,809 # info
2022-09-20 09:48:52,809 # mtd devices: 1
2022-09-20 09:48:52,809 #  -=[ MTD_0 ]=-
2022-09-20 09:48:52,810 # sectors: 3921920
2022-09-20 09:48:52,810 # pages per sector: 1
2022-09-20 09:48:52,810 # page size: 512
2022-09-20 09:48:52,810 # total: 1915 MiB
```

```
$ BOARD=remote-revb make -C tests/vfs_default all flash term
[...]
2022-09-20 09:49:37,168 # main(): This is RIOT! (Version: 2022.10-devel-680-gf03f5-remote-revb-mtd)
2022-09-20 09:49:37,168 # mount points:
2022-09-20 09:49:37,168 # 	/sd0
2022-09-20 09:49:37,168 # 
2022-09-20 09:49:37,168 # data dir: /sd0
> ls /sd0
2022-09-20 09:50:07,623 # ls /sd0
2022-09-20 09:50:07,631 # DATA/
2022-09-20 09:50:07,632 # LOG.TXT	32144 B
2022-09-20 09:50:07,634 # FIFO.BIN	45 B
2022-09-20 09:50:07,636 # FIDX.BIN	4 B
2022-09-20 09:50:07,639 # TIME.TXT	10 B
2022-09-20 09:50:07,640 # total 4 files
```

### CODE REVIEW

**Comment by benpicco** [comment]
This needs the same Kconfig fixes as #18445

**Comment by jdavid** [comment]
Done, rebased and squashed.

File: boards/remote-revb/Kconfig [hunk 1]
```diff
@@ -14,5 +14,13 @@ config BOARD_REMOTE_REVB
     select HAS_ARDUINO
     select HAVE_SAUL_ADC
     select HAVE_SAUL_GPIO
+    select HAS_SDCARD_SPI
```
Comment (benpicco) [inline boards/remote-revb/Kconfig:17]: `Makefile.features` now needs

    FEATURES_PROVIDED += sdcard_spi

Comment (jdavid) [reply boards/remote-revb/Kconfig:17]: done

**Review by benpicco — APPROVED** [review]

**Pull request merged**